pokemongo_bot/cell_workers/sniper.py

```diff
@@ -499,25 +448,50 @@ def _hash(self, pokemon):
+        for source in list(self.sources_pending_validation):
+            try:
+                if source.enabled:
+                    source.validate()
+                    self._log("Source '{}' is good! Storing...".format(source.url))
+                    self.sources_pending_validation.remove(source)
+                    self.sources.append(source)
+                else:
+                    self._log("Source '{}' is disabled! Ignoring...".format(source.url))
+                    self.sources_pending_validation.remove(source)
+            except LookupError as exception:
+                self._error("Source '{}' contains errors. Details: {}. Ignoring...".format(source.url, exception))
+                self.sources_pending_validation.remove(source)
+            except ValueError as exception:
+                self._error("Source '{}' could not be validated. Details: {}. Still pending...".format(source.url, exception))
+
+    def _is_enabled(self):
+        # Check if there are pending validations
+        if self.sources_pending_validation:
+            self._log("Validating sources: {}...".format(", ".join([source.url for source in self.sources_pending_validation])))
+            self._validate_sources()
+
+        # Check whether we have some snipe sources to snipe from or if catching has been disabled at all
+        if not self.sources and not self.sources_pending_validation:
+            self._error("There is no source available to snipe from. Skipping...")
+            return False
+        elif self.bot.catch_disabled:
+            # TODO: Show catching disabled warning GLOBALLY, only once
+            self._log("All catching tasks are currently disabled until {}. Sniper will resume when catching tasks are re-enabled".format(self.bot.catch_resume_at.strftime("%H:%M:%S")))
+            return False
+
+        return True
+
+    def _is_cached(self, pokemon):
+        return self._hash(pokemon) in self.bot.sniper_cache
 
-    def _cache(self, uniqueid):
-        if not self._is_cached(uniqueid):
-            # Free space if full and store it
+    def _cache(self, pokemon):
+        if not self._is_cached(pokemon):
+            # If cache is full, remove the first 25 registries (resource saving - too much comparisions)
             if len(self.bot.sniper_cache) >= self.MAX_CACHE_LIST_SIZE:
-                self.bot.sniper_cache.pop(0)
-            self.bot.sniper_cache.append(uniqueid)
-
-    def _build_unique_id(self, pokemon):
-        # Build unique id for this pokemon from id, latitude, longitude and expiration
-        uniqueid = str(pokemon.get('pokemon_id','')) + str(pokemon.get('latitude','')) + str(pokemon.get('longitude','')) + str(pokemon.get('expiration',''))
-        md5str = hashlib.md5()
-        md5str.update(uniqueid)
-        uniqueid = str(md5str.hexdigest())
-        return uniqueid
+                del self.bot.sniper_cache[:25]
+            self.bot.sniper_cache.append(self._hash(pokemon))
```

Comment:
You are passing the pokemon dict to _is_cached. But that's definitely an improvement over what you had before.